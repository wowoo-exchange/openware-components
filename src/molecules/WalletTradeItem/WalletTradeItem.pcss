.cr-wallet-trades-item {
  background-color: var(--wallet-trades-item-bg-color);
  color: var(--wallet-trades-item-color);
  display: flex;
  justify-content: space-between;

  &__body {
    display: inline-block;
    background-color: inherit;
    color: inherit;

    &-crypto {
      display: inline;
      background-color: inherit;
      justify-content: space-between;
      vertical-align: middle;
      color: inherit;

      &-icon {
        background-color: inherit;
        color: inherit;
        padding-left: calc(var(--gap) * 2);

        [class^=cr-crypto-font] {
          color: inherit;
          font-size: calc(var(--gap) * 6);
          vertical-align: middle;
          background: transparent;
        }
      }
    }

    &-data {
      display: inline-block;
      background-color: inherit;
      color: inherit;
      vertical-align: middle;

      &-currency {
        background-color: inherit;
        color: inherit;
        font-size: calc(var(--gap) * 2.33);
        padding-bottom: 5px;
      }

      &-balance {
        background-color: inherit;
        color: inherit;
        font-size: calc(var(--gap) * 2.66);
      }
    }
  }

  &__arrow {
    display: flex;
    float: right;
    margin-right: calc(var(--gap) * 4);
    line-height: calc(var(--gap) * 4.4);
    padding-top: calc(var(--gap) * 0.5);
    vertical-align: middle;
    background: transparent;
  }
}
