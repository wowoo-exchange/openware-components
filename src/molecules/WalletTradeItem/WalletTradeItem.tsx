import classnames from 'classnames';
import * as React from 'react';
import { CryptoIcon } from '../../atoms';
import iconRight = require('./chevron-right.svg');

interface WalletTradeItemProps {
    /**
     * Crypto currency code
     */
    currency: string;
    /**
     * Amount of currency
     */
    balance: number;
    /**
     * Additional class name for styling (by default `cr-wallet-trades-item`)
     */
    className?: string;
}

/**
 * Component for displaying information about wallet, including address and amount of currency.
 */
const WalletTradeItem: React.FunctionComponent<WalletTradeItemProps> = (props: WalletTradeItemProps) => {
    const {
        className,
        currency,
        balance,
    } = props;
    const cx = classnames('cr-wallet-trades-item', className);
    const currencyCode = `${currency.toUpperCase()}-alt`;
    return (
        <div className={cx}>
            <div className="cr-wallet-trades-item__body">
                <div className="cr-wallet-trades-item__body-crypto">
                    <CryptoIcon className="cr-wallet-trades-item__body-crypto-icon" code={currencyCode} />
                </div>
                <div className="cr-wallet-trades-item__body-data">
                    <div className="cr-wallet-trades-item__body-data-currency">
                        {currency.toUpperCase()}
                    </div>
                    <div className="cr-wallet-trades-item__body-data-balance">
                        {balance}
                    </div>
                </div>
            </div>
            <div className="cr-wallet-trades-item__arrow">
                <img src={iconRight} />
            </div>
        </div>
    );
};

export {
    WalletTradeItem,
    WalletTradeItemProps,
};
