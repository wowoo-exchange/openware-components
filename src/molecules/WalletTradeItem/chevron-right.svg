<svg width="9" height="14" viewBox="0 0 9 14" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M0.0078125 12.3594L5.36719 7L0.0078125 1.64062L1.64844 0L8.64844 7L1.64844 14L0.0078125 12.3594Z" fill="white"/>
</svg>
