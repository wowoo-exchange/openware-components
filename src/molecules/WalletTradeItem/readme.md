WalletTradeItem example:

```js
<WalletTradeItem currency="btc" balance={1.5501} />
```
