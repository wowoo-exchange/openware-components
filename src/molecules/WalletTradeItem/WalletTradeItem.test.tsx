import { shallow } from 'enzyme';
import * as React from 'react';
import { WalletTradeItem, WalletTradeItemProps } from './WalletTradeItem';

const defaultProps: WalletTradeItemProps = {
    balance: 0,
    currency: 'btc',
};

const setup = (props: Partial<WalletTradeItemProps> = {}) =>
    shallow(<WalletTradeItem {...{ ...defaultProps, ...props }} />);

describe('WalletTradeItem', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('should have correct className', () => {
        const wrapper = setup();
        expect(wrapper.hasClass('cr-wallet-trades-item')).toBeTruthy();
    });

    it('should pass along supplied className', () => {
        const className = 'some-class';
        const wrapper = setup({ className });
        expect(wrapper.hasClass(className)).toBeTruthy();
    });
});
