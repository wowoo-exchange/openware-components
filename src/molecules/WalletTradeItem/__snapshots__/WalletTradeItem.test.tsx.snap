// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletTradeItem should render 1`] = `
<div
  className="cr-wallet-trades-item"
>
  <div
    className="cr-wallet-trades-item__body"
  >
    <div
      className="cr-wallet-trades-item__body-crypto"
    >
      <CryptoIcon
        className="cr-wallet-trades-item__body-crypto-icon"
        code="BTC-alt"
      />
    </div>
    <div
      className="cr-wallet-trades-item__body-data"
    >
      <div
        className="cr-wallet-trades-item__body-data-currency"
      >
        BTC
      </div>
      <div
        className="cr-wallet-trades-item__body-data-balance"
      >
        0
      </div>
    </div>
  </div>
  <div
    className="cr-wallet-trades-item__arrow"
  >
    <img
      src="test-file-stub"
    />
  </div>
</div>
`;
